ops/testing.py

```diff
@@ -140,7 +140,7 @@ def _next_relation_id(self):
         self._relation_id_counter += 1
         return rel_id
 
-    def add_relation(self, relation_name, remote_app, remote_app_data=None):
+    def add_relation(self, relation_name, remote_app, *, remote_app_data=None):
```

Comment:
Is this change tested? (same with similar change below)

(BTW, I think it's a good idea to force kwargs parameters when it's too confusing to be used as positional)

Reply:
I didn't add a test that "you cannot pass this as an *arg". I did find that in all the places I was using it, I was using a named arg because it felt a lot cleaner, so I wanted to recommend that to others. If you feel strongly I can have the:
```
with self.assertRaises(TypeError) as cm:
  harness.add_relation('name', 'app', {'app': 'data'})
```
it doesn't seem like the sort of test I would have added if I had written the code that way from the beginning, though.

Reply:
Is this change providing any value to the code? If yes, it should be tested so it's not accidentally lost in a refactor.

This is particularly important as it's an interface exposed to the developer, so we need to handle this carefully (for example, this kind of change can be made now without handling interface evolution because we're alpha, but in a future this kind of change will be need to be managed differently).

Reply:
added tests for these